docs/autobridge.rst

```diff
@@ -0,0 +1,199 @@
+- To resolve global critical paths, AutoBridge adds additional pipelines to the latency-insensitive interfaces between tasks.
+
+
+.. image:: https://user-images.githubusercontent.com/32432619/165636025-a85940ac-70f9-4a2d-8376-c1a96510e449.png
+  :width: 50 %
+
+Manipulate the Floorplan Parameters
+::::::::::::::::::::::::::::::::::::::::
+
+The floorplanning process is about the tradeoff between two factors:
+
+- Area limit: the maximal percentage of occupied resources in each slot
+- SLR crossing limit: the maximal number of wires crossing an SLR boundary
+
+Given a specific area limit and an SLR crossing limit, AutoBridge assigns each task to one slot such that minimize the total wire length.
+By default, AutoBridge will empirically select an area limit and an SLR crossing limit within an acceptable range. 
+
+You could manipulate AutoBridge through the following options:
+
+- ``--min-area-limit`` and ``--max-area-limit`` specify the range of acceptable area limit that AutoBridge chooses from.
+- ``--min-slr-width-limit`` and ``--max-slr-width-limit`` specify the range of acceptable SLR crossing limit that AutoBridge chooses from.
+- ``--floorplan-opt-priority`` controls the priority between the two factors. By default, AutoBridge takes priority in selecting a smaller area limit (``AREA_PRIORITIZED``). You could change it by providing the ``SLR_CROSSING_PRIORITIZED`` option. 
+
+Manual Floorplanning
+:::::::::::::::::::::::
+
+You could enforce the tool to assign certain task instances to specified slots. To do this, use the ``--floorplan-pre-assignments`` option to provide a json file that includes the manual floorplanning.
+
+- A common use case is to reproduce the floorplan step of a previous run. In each time, AutoBridge will generate a ``floorplan-region-to-instances.json`` that records the entire floorplan results. If you simply provide this file from a previous run, then you are essentially reproducing the previous run.
+
+- This step requires you to know some lower level information of the device.
+
+- Here is an example json file. ``CR_X0Y0_To_CR_X3Y3`` stands for "the rectangle region determined by CLOCKREGION_X0Y0 and CLOCKREGION_X3Y3". Assume that the task ``foo`` is invoked twice, this json assigns the first invoked instance ``TASK_VERTEX_foo_1`` to ``CR_X0Y0_To_CR_X3Y3`` and the second invoked instance to ``CR_X4Y8_To_CR_X7Y11``.
+
+.. code-block:: json
+
+  {
+    "CR_X0Y0_To_CR_X3Y3": [
+      "TASK_VERTEX_foo_1",
+      "TASK_VERTEX_bar"
+    ],
+    "CR_X4Y8_To_CR_X7Y11": [
+      "TASK_VERTEX_foo_2",
+      "TASK_VERTEX_taz"
+    ]
+  }
+
+Scalability
+::::::::::::
+
+AutoBridge has been tested on designs from <20 tasks to designs with >1000 tasks. Here are some tips if you have more than hundreds of tasks:
+
+- Install the Gurobi solver. Gurobi is free for academia, and the registration process takes less than 5 minutes. See instructions `here <https://github.com/UCLA-VAST/tapa/blob/master/docs/installation.rst>`_.
```

Comment:
Let's not link to the source code. Those source code are not for human eyes. Instead, do ``:ref:`here <installation:install gurobi (recommended)>` ``